File: src/p485MaxConsecutiveOnes/Solution.java
package p485MaxConsecutiveOnes;

public class Solution {

	public int findMaxConsecutiveOnes(int[] nums) {
		int res = 0;
		int tmp = 0;
		for(int i = 0; i < nums.length; i ++){
			if(nums[i] == 1){
				tmp ++;
				res = Math.max(res, tmp);
			}else{
				tmp = 0;
			}
		}
		return res;
	}

	public static void main(String[] args) {
		System.out.println(new Solution().findMaxConsecutiveOnes(new int[]{1,1,0,1,1,1}));
	}

}

File: src/p485MaxConsecutiveOnes/solution.py
# -*- coding: utf-8 -*-
# @Time    : 2018/8/31 11:00
# @Author  : yunfan

class Solution:
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l, r = 0, 0
        res = 0
        while r < len(nums):
            if nums[r] == 1:
                if r > 0 and nums[r - 1] != 1:
                    l = r
                res = max(res, r - l + 1)
            r += 1
        return res